### PR TITLE
Comments presentation

### PR DESCRIPTION
**Pull request by Fourchaux**
Continuation of https://github.com/c-cube/ocaml-containers/pull/300

If CCList is OK, A PR for _CCListLabels.mli_ will follow...

### CODE REVIEW

File: src/core/CCList.mli [hunk 1]
```diff
@@ -24,74 +24,75 @@ val is_empty : _ t -> bool
...
 
 val cons_maybe : 'a option -> 'a t -> 'a t
 (** [cons_maybe (Some x) l] is [x :: l].
     [cons_maybe None l] is [l].
     @since 0.13 *)
 
 val (@) : 'a t -> 'a t -> 'a t
-(** Like [append].
-    Concatenate two lists. *)
+(** [l1 @ l2] is like [append].
```
Comment (FardaleM) [inline src/core/CCList.mli:44]: Maybe replace `[append]` by `[append l1 l2]`.

File: src/core/CCList.mli [hunk 2]
```diff
@@ -24,74 +24,75 @@ val is_empty : _ t -> bool
...
 
 val fold_while : ('a -> 'b -> 'a * [`Stop | `Continue]) -> 'a -> 'b t -> 'a
-(** Fold until a stop condition via [('a, `Stop)] is
+(** [fold_while f acc l] folds until a stop condition via [('a, `Stop)] is
     indicated by the accumulator.
     @since 0.8 *)
 
 val fold_map : ('acc -> 'a -> 'acc * 'b) -> 'acc -> 'a list -> 'acc * 'b list
-(** [fold_map f init l] is a [fold_left]-like function, but it also maps the
+(** [fold_map f acc l] is a [fold_left]-like function, but it also maps the
```
Comment (FardaleM) [inline src/core/CCList.mli:64]: Why did you rename `init` to `acc`? I, personally, see this value as the initial value of the fold.

File: src/core/CCList.mli [hunk 3]
```diff
@@ -111,73 +112,78 @@ val count : ('a -> bool) -> 'a list -> int
...
-(** Map with index and flatten at the same time (safe).
+(** [flat_map_i f l] maps with index and flattens at the same time (safe).
     Evaluation order is not guaranteed.
-    @since 2.8
-*)
+    @since 2.8 *)
 
 val flatten : 'a t t -> 'a t
-(** Safe flatten. Concatenate a list of lists. *)
+(** [flatten [l1]; [l2]; ⋯] concatenates a list of lists. Safe flatten. *)
```
Comment (FardaleM) [inline src/core/CCList.mli:176]: Maybe you can use the same notation as before `Safe version of {!List.flatten}`

File: src/core/CCList.mli [hunk 4]
```diff
@@ -302,18 +308,18 @@ val sublists_of_len :
...
 
 val intersperse : 'a -> 'a list -> 'a list
-(** Insert the first argument between every element of the list.
+(** [intersperse x l] inserts the element [x] between adjacent elements of the list [l].
     @since 2.1, but only
     @since 2.2 with labels *)
 
 val interleave : 'a list -> 'a list -> 'a list
-(** [interleave [x1…xn] [y1…ym]] is [x1,y1,x2,y2,…] and finishes with
+(** [interleave [x1…xn] [y1…ym]] is [x1;y1;x2;y2;…] and finishes with
```
Comment (FardaleM) [inline src/core/CCList.mli:316]: The `[` and `]` will not be print in the documentation for `[x1;y1;x2;y2;…]`, if you want to have the OCaml representation of a list you should add an extra `[]` around it.

File: src/core/CCList.mli [hunk 5]
```diff
@@ -360,15 +366,16 @@ val last : int -> 'a t -> 'a t
     [l] doesn't have that many elements). *)
 
 val head_opt : 'a t -> 'a option
-(** First element.
+(** [head_opt (x :: l)] returns [Some x] or [None] if the list is empty.
```
Comment (FardaleM) [inline src/core/CCList.mli:369]: `(x :: l)` may give the impression that head_opt should be call with a non-empty list only.

File: src/core/CCList.mli [hunk 6]
```diff
@@ -435,33 +442,35 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
 val sorted_merge_uniq : cmp:('a -> 'a -> int) -> 'a list -> 'a list -> 'a list
-(** [sorted_merge_uniq l1 l2] merges the sorted lists [l1] and [l2] and
+(** [sorted_merge_uniq ~cmp l1 l2] merges the sorted lists [l1] and [l2] and
     removes duplicates.
     @since 0.10 *)
 
 val is_sorted : cmp:('a -> 'a -> int) -> 'a list -> bool
-(** [is_sorted l] returns [true] iff [l] is sorted (according to given order).
+(** [is_sorted ~cmp l] returns [true] iff [l] is sorted (according to given order).
     @param cmp the comparison function (default [Stdlib.compare]).
```
Comment (FardaleM) [inline src/core/CCList.mli:459]: `cmp` is a mandatory argument, `(default [Stdlib.compare])` should be removed.

File: src/core/CCList.mli [hunk 7]
```diff
@@ -576,73 +595,84 @@ val range : int -> int -> int t
     both for decreasing and increasing ranges. *)
 
 val range' : int -> int -> int t
-(** Like {!range} but the second bound is excluded.
+(** [range' i j] is like {!range} but the second bound [j] is excluded.
     For instance [range' 0 5 = [0;1;2;3;4]]. *)
 
 val (--) : int -> int -> int t
-(** Infix alias for [range]. *)
+(** [i -- j] iterates on integers from [i] to [j] included.
```
Comment (FardaleM) [inline src/core/CCList.mli:602]: `(--)` does not really iterate, it creates a list.

File: src/core/CCList.mli [hunk 8]
```diff
@@ -360,15 +366,16 @@ val last : int -> 'a t -> 'a t
...
 
 val head_opt : 'a t -> 'a option
-(** First element.
+(** [head_opt (x :: l)] returns [Some x] or [None] if the list is empty.
     @since 0.20 *)
 
 val tail_opt : 'a t -> 'a t option
-(** Return the given list without its first element.
+(** [tail_opt (x :: l)] returns [Some l] (the given list without its first element),
+    or [None] if the list is empty.
```
Comment (FardaleM) [inline src/core/CCList.mli:374]: Same remark as for `head_opt`.

File: src/core/CCList.mli [hunk 9]
```diff
@@ -576,73 +595,84 @@ val range : int -> int -> int t
...
     For instance [range' 0 5 = [0;1;2;3;4]]. *)
 
 val (--) : int -> int -> int t
-(** Infix alias for [range]. *)
+(** [i -- j] iterates on integers from [i] to [j] included.
+    Infix alias for [range]. *)
 
 val (--^) : int -> int -> int t
-(** Infix alias for [range'].
+(** [i --^ j] iterates on integers from [i] to [j] excluded.
```
Comment (FardaleM) [inline src/core/CCList.mli:606]: Same remark as for `(--)`

File: src/core/CCList.mli [hunk 10]
```diff
@@ -703,51 +736,51 @@ val random_non_empty : 'a random_gen -> 'a t random_gen
...
 
 val of_std_seq_rev : 'a Seq.t -> 'a t
-(** Build a list from a given [Seq.t], in reverse order.
+(** [of_std_seq_rev Seq.t] builds a list from a given [Seq.t], in reverse order.
     @since 2.8 *)
 
 val of_std_seq : 'a Seq.t -> 'a t
-(** Build a list from a given [Seq.t].
-    In the result, elements appear in the same order as they did in the source [seq].
+(** [of_std_seq Seq.t] builds a list from a given [Seq.t].
```
Comment (FardaleM) [inline src/core/CCList.mli:768]: `Seq.t` is a strange name for an argument.

File: src/core/CCList.mli (hunk 1 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: OK.

File: src/core/CCList.mli (hunk 3 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: OK.

File: src/core/CCList.mli (hunk 4 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: OK.

File: src/core/CCList.mli (hunk 5 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: Modified.

File: src/core/CCList.mli (hunk 6 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: OK.

File: src/core/CCList.mli (hunk 7 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: Modified.

File: src/core/CCList.mli (hunk 8 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: Modified.

File: src/core/CCList.mli (hunk 9 above)
Comment (Fourchaux) [reply src/core/CCList.mli]: Modified.

File: src/core/CCList.mli (hunk 10 above)
Comment (Fourchaux) [reply src/core/CCList.mli:770]: Seq.t is used in other modules. (CCArray, CCString ...).
For these names (and for now) I suggest keeping them as is. At the end of 'Phase One' 
(this reformatting of comments ), we can establish a complete index of the names and modify them eventually.

Comment (FardaleM) [reply src/core/CCList.mli:770]: OK, then can you remove the `Seq.t` from `of_std_seq Seq.t` which is not correct OCaml?

Comment (c-cube) [reply src/core/CCList.mli:770]: must be a search and replace that got out of hand :)

Comment (Fourchaux) [reply src/core/CCList.mli:770]: @FardaleM  _...which is not correct OCaml_

Sorry, but I don't understand this remark.
- We talk about _Seq.t_ used in COMMENTS
- And even in OCaml code, maybe _Seq.t_ is strange but completely correct.
Tons of programs use it: OCaml stdlib, alt-ergo, lwt, bap, fmt,coccinelle and so on.
 
Anyway, to sum up the situation: 
if I use **seq** instead of **Seq.t** is it OK for you?

Comment (FardaleM) [reply src/core/CCList.mli:770]: Yes it is fine for me.
To explain my remark, `[]` is often used to print code in the doc and `of_std_seq Seq.t` looks very strange but I made a mistake, it can be correct in OCaml.

Comment (FardaleM) [reply src/core/CCList.mli:770]: You can change it or leave it as you prefer. If it is in other modules as well we need to change all of them at some point.

**Comment by Fourchaux** [comment]
All 'Seq.t' replaced by 'seq' (in the src/core modules).

Thank you.

**Comment by FardaleM** [comment]
Perfect!
For the CCListLabels.mli do you want to add it in this PR or I merge it and you open a new one when it is ready?

**Comment by Fourchaux** [comment]
Merge this PR please.
I'll open a new one for CCListLabels.mli.

TY
